packages/react-data-grid/src/ToolbarContainer.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { isElement, isValidElementType } from 'react-is';
+import { Column } from './common/types';
+
+interface ToolbarProps {
+  columns: Column[];
+  rowsCount: number;
+  onToggleFilter(): void;
+}
+
+export interface ToolbarContainerProps extends ToolbarProps {
+  /** Component used to render toolbar above the grid */
+  toolbar: React.ReactElement<ToolbarProps> | React.ComponentType<ToolbarProps>;
```

Comment:
Should we change the type to 

```
toolbar: React.ReactElement<ToolbarProps> | React.ComponentType<ToolbarProps>;
```

or change the call to 

```
this.props.toolbar && <ToolbarContainer
```

Reply:
I think we should strive to reduce the complexity in big components, hence why I make this ToolbarContainer in the first place, so I went with your first suggestion.